src/core/Microsoft.PowerToys.Settings.UI/ViewModels/GeneralViewModel.cs

```diff
@@ -2,14 +2,225 @@
 // The Microsoft Corporation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
 using Microsoft.PowerToys.Settings.UI.Helpers;
+using Microsoft.PowerToys.Settings.UI.Lib;
+using Microsoft.PowerToys.Settings.UI.ViewModels.Commands;
+using Microsoft.PowerToys.Settings.UI.Views;
+using Windows.System;
+using Windows.UI.Popups;
+using Windows.UI.Xaml;
 
 namespace Microsoft.PowerToys.Settings.UI.ViewModels
 {
     public class GeneralViewModel : Observable
     {
+        private GeneralSettings GeneralSettingsConfigs { get; set; }
+
+        public ButtonClickCommand CheckFoUpdatesEventHandler { get; set; }
+
+        public ButtonClickCommand RestartElevatedButtonEventHandler { get; set; }
```

Comment:
[nit] Maybe it would be better if this properties were declared by their interface, `ICommand`. Also, I think it might be simpler to use the Click event instead of `ICommand` for the button actions, since the `CanExecute` method is not really being used.

Reply:
I dont think we should use the Click Event because the CanExecute is not being used. The ICommand is used to have all the logic in one ViewModel file.